zerver/lib/create_user.py

```diff
@@ -127,4 +132,14 @@ def create_user(email: str, password: Optional[str], realm: Realm,
     user_profile.save(update_fields=["recipient"])
 
     Subscription.objects.create(user_profile=user_profile, recipient=recipient)
+
+    if user_avatar_url:
+        from zerver.lib.upload import upload_avatar_image_from_url
+        from zerver.lib.actions import do_change_avatar_fields
```

Comment:
I think we can put the imports at the top of the file, the usual way, or do we have a reason to do local import like this?

Reply:
I saw that in other parts of the code, people were doing local import when is something that will only be used by the local scope, so I tried to use the same pattern as those functions are only used here. Should I move them to the top?

Reply:
My understanding is that we do it in 2 cases:
1. Sometimes it's necessary to avoid circular imports.
2. If the the import is really needed in one place that's called rarely enough and the import is expensive enough to want to avoid importing it on module level and just do it on demand, in the function that needs. Example would be `import talon` in `email_mirror.py`.

Otherwise I think we just put it at the top of the value. Maybe @timabbott will have some different comment on this.

Reply:
Something odd happened here. I tried to move those imports to the top and it causes an import error. Also, there is another snippet of code that imports `do_change_avatar_fields` locally.

![Captura de tela de 2020-03-23 18-05-32](https://user-images.githubusercontent.com/15564640/77363332-49f77280-6d31-11ea-9fba-007a6a28ab93.png)

Reply:
Ah, that's exactly what happens with circular imports! Seem like it was right to import locally after all :slightly_smiling_face: